lassen/mode.py

```diff
@@ -10,27 +10,28 @@ class Mode_t(Enum):
     BYPASS = 2  # Register is bypassed and input value is returned
     DELAY = 3   # Register written with input value, previous value returned
 
-@lru_cache(None)
-def gen_register_mode(T, init=0):
-    @family_closure
-    def RegisterMode_fc(family):
-        T_f = rebind_type(T, family)
-        Reg = gen_register(T_f, init)(family)
+@family_closure
```

Comment:
Why not keep it as generator returning a family closure? switching (FC produceing a generator) might break some of the auto-mapping code

Reply:
Because it required the sort backward rebind thing which in my mind was a bit dumb.   Its not technically necessary but I doubt it will break auto mapping code unless it is targeting the registermode directly. Another option would be to structure the code like:
```Python
def gen_mode(width, init):
    @fc
    def RegisterMode_fc(fam):
        T = fam.BitVector[width]
        Reg = fam.gen_register(T, init)
        ...
```

Reply:
But I an revert if it breaks things

Reply:
I would prefer the code be structured like that if you do not like the rebind